apps/dotcom/client/src/tla/components/TlaEditor/TlaEditor.tsx

```diff
@@ -163,7 +156,35 @@ function TlaEditorInner({
 				}
 				updateSessionState(state)
 			})
+
+			const abortController = new AbortController()
+			if (app._slurpFileId === fileId) {
+				// This is a one-time operation.
+				// So we need to wait a tick for react strict mode to finish
+				// doing its nasty business before we start the migration.
+				sleep(50)
+					.then(async () => {
+						if (!abortController.signal.aborted) {
+							if (abortController.signal.aborted) return
+							await slurpLocalContent(editor, abortController.signal)
+							if (abortController.signal.aborted) return
+							uploadLegacyFiles(editor, abortController.signal)
+							app._slurpFileId = null
+							setIsReady()
+						}
+					})
+					.catch((_e) => {
+						// show migration error
+						// give people link to /local and tell them to
+						// create a tldr file and drag it in
+					})
+			} else {
+				uploadLegacyFiles(editor, abortController.signal)
```

Comment:
I guess this is retry logic in case loading of some assets in the `if` block failed? We run this for all the files though, even when they have nothing to do with legacy files.

Reply:
yep it will stop immediately if there's no slurpPersistenceKey though